python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -68,6 +80,17 @@ def __init__(self, registryConfig, schemaConfig):
         self._engine = create_engine(self.config['db'])
         self._schema.metadata.create_all(self._engine)
         self._datasetTypes = {}
+        self._connection = self._engine.connect()
+
+    @contextlib.contextmanager
+    def transaction(self):
+        trans = self._connection.begin()
+        try:
+            yield
+            trans.commit()
+        except Exception:
```

Comment:
Are we sure that this should not be `BaseException`? If I do a Ctrl-C what happens?